### PR TITLE
Less verbose progress reports

### PR DESCRIPTION
**Pull request by jasmainak**
cc @jona-sassenhagen let's see if this makes you happy. I think it should work with ipython notebooks. But I haven't tried yet. It should be drastically less verbose. It's another dependency but it's light and I think worth it.

I can add verbosity parameter too if you insist ;) But I might need more time to figure it out in that case ...

### CODE REVIEW

File: autoreject/autoreject.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 from scipy.stats.distributions import uniform
 
 import mne
-from mne.utils import ProgressBar
+from tqdm import tqdm, trange
```
Comment (agramfort) [inline autoreject/autoreject.py]: are you sure you want this to be a hard dependency?

Comment (jasmainak) [inline autoreject/autoreject.py]: lol I didn't expect to see you here @agramfort :) okay you got me thinking. Maybe I'll add a fallback to the `ProgressBar` from `mne.utils` ...

**Comment by agramfort** [comment]
Big brother is watching you ;)

**Comment by jasmainak** [comment]
@jona-sassenhagen: 
output1
:) I'll merge when tests are happy and you can try it!

**Comment by mmagnuski** [comment]
that's a really cool progressbar :)

**Comment by jona-sassenhagen** [comment]
Uh oh ...

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-17-1b45e517c461> in <module>()
      5     eps, _ = get_epochs(subj)
      6     ar = LocalAutoRejectCV(verbose=False)
----> 7     ar.fit(eps)
      8 
      9     with open('/home/jona/experiments/psc/ar/{}-ar.pckl'.format(name), 'wb') as f:

/home/jona/tools/autoreject/autoreject/autoreject.py in fit(self, epochs)
    532 
    533         # The thresholds must be learnt from the entire data
--> 534         local_reject.fit(epochs)
    535 
    536         local_reject._vote_epochs(epochs)

/home/jona/tools/autoreject/autoreject/autoreject.py in fit(self, epochs)
    340             estimated.
    341         """
--> 342         self.threshes_ = self.thresh_func(epochs)
    343         return self
    344 

/home/jona/tools/autoreject/autoreject/autoreject.py in compute_thresholds(epochs, method, random_state, verbose)
    260                 if ch_type in epochs]
    261     n_epochs = len(epochs)
--> 262     epochs_interp = clean_by_interp(epochs, verbose=verbose)
    263     data = np.concatenate((epochs.get_data(), epochs_interp.get_data()),
    264                           axis=0)

/home/jona/tools/autoreject/autoreject/utils.py in clean_by_interp(inst, verbose)
     82     ch_names = [ch_name for ch_name in inst.info['ch_names']]
     83     for ch_idx, (pick, ch) in enumerate(_pbar(zip(picks, ch_names), desc=mesg,
---> 84                                         verbose=verbose)):
     85         inst_clean = inst.copy()
     86         inst_clean.info['bads'] = [ch]

/home/jona/tools/autoreject/autoreject/utils.py in _pbar(iterable, desc, leave, position, verbose)
     44                     self.update_with_increment_value(1)
     45 
---> 46         pbar = _ProgressBar(iterable, mesg=desc, spinner=True)
     47     elif verbose == 'tqdm':
     48         from tqdm import tqdm

/home/jona/tools/mne-python/mne/utils.py in __init__(self, max_value, initial_value, mesg, max_chars, progress_character, spinner, verbose_bool)
   1668         else:
   1669             # input is an iterable
-> 1670             self.max_value = len(max_value)
   1671             self.iterable = max_value
   1672         self.mesg = mesg

TypeError: object of type 'zip' has no len()
```

I get that for all progressbar options, even `False`.

**Comment by jasmainak** [comment]
wait, what version of `mne` are you using?

**Comment by jona-sassenhagen** [comment]
0.14.dev0. Pulled yesterday or so.

**Comment by jasmainak** [comment]
and you're using the latest version of `autoreject` after the merge?

**Comment by jona-sassenhagen** [comment]
Yes. Just pulled autoreject and mne.

**Comment by jona-sassenhagen** [comment]
Works with b8f51139809bb29e6274857c5450ebdc39a60726 fwiw

**Comment by jasmainak** [comment]
I think I found the problem ... pushing a fix. One moment

**Comment by jasmainak** [comment]
okay try now!

**Comment by jona-sassenhagen** [comment]
verbose=False works now, but the default stills gives me the same error ...

**Comment by jasmainak** [comment]
I'm mystified. If you do:

```py
>>> len(zip([2, 3], [1, 2]))
```
do you get an error? Is it python 2.7 you are using?

**Comment by mmagnuski** [comment]
On py3 you need to use list on zip, otherwise it has no len... such is the zen ;)

**Comment by jona-sassenhagen** [comment]
Aha!

```
Python 3.5.2 |Anaconda custom (64-bit)| (default, Jul  2 2016, 17:53:06)
Type "copyright", "credits" or "license" for more information.

IPython 5.1.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: len(zip([2, 3], [1, 2]))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-0d2fb0f8fdd9> in <module>()
----> 1 len(zip([2, 3], [1, 2]))

TypeError: object of type 'zip' has no len()
```

**Comment by jasmainak** [comment]
So, I suppose this calls for a fix here: https://github.com/mne-tools/mne-python/blob/master/mne/utils.py#L1670 ? Or Jona, if you can switch to py 2.7, that's probably the easier route ...

**Comment by jasmainak** [comment]
... and don't forget you can use the fancy progressbars from `tqdm` too while staying in python 3. I'm wondering if there's an easy fix via `six` ...

**Comment by jasmainak** [comment]
never mind, [this commit](https://github.com/autoreject/autoreject/commit/bacfe1496cc684ff94d53e21b66a1d48b360664b) should fix it

**Pull request merged**